openeo_processes_dask/process_implementations/cubes/resample.py

```diff
@@ -49,14 +52,40 @@ def resample_spatial(
             f'Selected resampling method "{method}" is not available! Please select one of '
             f"[{', '.join(resample_methods_list)}]"
         )
+    
+    dims = data.dims
+    if dims is None:
+        raise OpenEOException(
+            f"Data cube has no dimensions"
+        )
 
-    # Re-order, this is specifically done for odc reproject
-    data_cp = data.transpose(
-        data.openeo.band_dims[0],
-        data.openeo.temporal_dims[0],
-        data.openeo.y_dim,
-        data.openeo.x_dim,
+    try:
+        if 'band' in dims and 'time' in dims and 'y' in dims and 'x' in dims:
+            # all dimensions present
+            data_cp = data.transpose('band', 'time', 'y', 'x')
+        elif 'time' in dims and 'y' in dims and 'x' in dims:
+            # no band dimension
+            data_cp = data.transpose('time', 'y', 'x')
+        elif 'band' in dims and 'y' in dims and 'x' in dims:
+            # no time dimension
+            data_cp = data.transpose('band', 'y', 'x')
+        else:
+            # no band and time dimension
+            data_cp = data.transpose('y', 'x')
+    except Exception: 
+        raise OpenEOException(
+        f"Data cube has unexpected dimensions: {dims}"
```

Comment:
Thanks for your inputs on the process and adding the align parameter!
I think, we can keep using the `data.openeo` here, as this checks the dimensions in the dataset - and for example uses `t` instead of `time` if the temporal dimension is called like that.

Reply:
Hi, sure but using data.openeo limits you to using `band` , `time`, `x` and `y` only now no? Because the moment you add a dimension and run resample it fails. We are looking into ways to extend the support of other dimensions. We can certainly reuse data.openeo for something but ultimately extend the `resample_spatial` dim support.

Reply:
we could do something like this maybe:

```Python
    dims = list(data.dims)

    known_dims = [
        data.openeo.band_dims[0],
        data.openeo.temporal_dims[0],
        data.openeo.y_dim,
        data.openeo.x_dim,
    ]

    other_dims = [dim for dim in dims if dim not in known_dims]

    data_cp = data.transpose(*other_dims, *known_dims)
```

This way we can keep the `data.openeo` features plus add other dimensions. Of course, some limits and checks would need to be performed on the `other` dimensions as we cannot realistically allow and support everything.

Reply:
Hi Valentina!

I updated the latest commit based on your suggestion to keep the `data.openeo` functionality. The dummy align is set as well to prevent errors from occurring. Let me know what you think!

Reply:
LGTM! 

The tests are currently failing because you need to `pre-commit run --all-files` on the branch.